File: packages/plugin-simplicity/fixtures/go/store.go
package fixtures

import (
	"errors"
	"io/fs"
	"os"
)

func ReadStore(path string) string {
	text, err := os.ReadFile(path)
	if isMissing(err) {
		return ""
	}
	return string(text)
}

func isMissing(err error) bool {
	return errors.Is(err, fs.ErrNotExist)
}

File: packages/plugin-simplicity/fixtures/rust/store.rs
use std::io;

pub fn read_store(path: &str) -> String {
    match std::fs::read_to_string(path) {
        Ok(text) => text,
        Err(error) if is_missing(&error) => String::new(),
        Err(error) => panic!("{}", error),
    }
}

fn is_missing(error: &io::Error) -> bool {
    error.kind() == io::ErrorKind::NotFound
}
